src/components/App.js

```diff
@@ -2,26 +2,43 @@ import React, {Component} from 'react';
 import Header from './Header';
 import MovieContainer from './MovieContainer';
 import MovieView from './MovieView';
-import movieData from '../movieData';
+import apiCalls from '../apiCalls';
+import { cleanMovieData, formatAverageRating } from "../utilities.js";
 import '../css/App.css';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData,
+      movies: [],
       error: "",
       isLoading: true,
       selectedMovie: ''
     };
   }
 
+  componentDidMount = () => {
+    apiCalls.getMovieData()
+      .then((response) => response.json())
+      .then(({ movies }) => {
+        const cleanedMoviePoster = movies.map((movie) => {
+          const formattedRating = formatAverageRating(movie["average_rating"]);
+          return { ...movie, average_rating: formattedRating };
+        });
+        this.setState({ movies: cleanedMoviePoster });
+      })
+      .catch((error) => this.setState({ error: error.message }));
+  };
+
   handleClick = (id) => {
```

Comment:
Be more specific with your naming of `handeClick` because you might have a lot of click handlers in your application. As you pass those around through props, the meaning is lost, and you will then need to go find the handler function to find out more about it.